homeassistant/components/deconz/device_automation.py

```diff
@@ -0,0 +1,255 @@
+    (CONF_DOUBLE_PRESS, CONF_RIGHT): 2004,
+    (CONF_SHORT_PRESS, CONF_BOTH_BUTTONS): 3002,
+    (CONF_LONG_PRESS, CONF_BOTH_BUTTONS): 3001,
+    (CONF_DOUBLE_PRESS, CONF_BOTH_BUTTONS): 3004,
+}
+
+AQARA_MINI_SWITCH_MODEL = "lumi.remote.b1acn01"
+AQARA_MINI_SWITCH = {
+    (CONF_SHORT_PRESS, CONF_TURN_ON): 1002,
+    (CONF_DOUBLE_PRESS, CONF_TURN_ON): 1004,
+    (CONF_LONG_PRESS, CONF_TURN_ON): 1001,
+    (CONF_LONG_RELEASE, CONF_TURN_ON): 1003,
+}
+
+AQARA_ROUND_SWITCH_MODEL = "lumi.sensor_switch"
+AQARA_ROUND_SWITCH = {
+    (CONF_SHORT_PRESS, CONF_TURN_ON): 1000,
+    (CONF_SHORT_RELEASE, CONF_TURN_ON): 1002,
+    (CONF_DOUBLE_PRESS, CONF_TURN_ON): 1004,
+    (CONF_TRIPLE_PRESS, CONF_TURN_ON): 1005,
+    (CONF_QUADRUPLE_PRESS, CONF_TURN_ON): 1006,
+    (CONF_QUINTUPLE_PRESS, CONF_TURN_ON): 1010,
+    (CONF_LONG_PRESS, CONF_TURN_ON): 1001,
+    (CONF_LONG_RELEASE, CONF_TURN_ON): 1003,
+}
+
+AQARA_SQUARE_SWITCH_MODEL = "lumi.sensor_switch.aq3"
+AQARA_SQUARE_SWITCH = {
+    (CONF_SHORT_PRESS, CONF_TURN_ON): 1002,
+    (CONF_DOUBLE_PRESS, CONF_TURN_ON): 1004,
+    (CONF_LONG_PRESS, CONF_TURN_ON): 1001,
+    (CONF_LONG_RELEASE, CONF_TURN_ON): 1003,
+    (CONF_SHAKE, ""): 1007,
+}
+
+REMOTES = {
+    HUE_DIMMER_REMOTE_MODEL: HUE_DIMMER_REMOTE,
+    HUE_TAP_REMOTE_MODEL: HUE_TAP_REMOTE,
+    TRADFRI_ON_OFF_SWITCH_MODEL: TRADFRI_ON_OFF_SWITCH,
+    TRADFRI_OPEN_CLOSE_REMOTE_MODEL: TRADFRI_OPEN_CLOSE_REMOTE,
+    TRADFRI_REMOTE_MODEL: TRADFRI_REMOTE,
+    TRADFRI_WIRELESS_DIMMER_MODEL: TRADFRI_WIRELESS_DIMMER,
+    AQARA_DOUBLE_WALL_SWITCH_MODEL: AQARA_DOUBLE_WALL_SWITCH,
+    AQARA_MINI_SWITCH_MODEL: AQARA_MINI_SWITCH,
+    AQARA_ROUND_SWITCH_MODEL: AQARA_ROUND_SWITCH,
+    AQARA_SQUARE_SWITCH_MODEL: AQARA_SQUARE_SWITCH,
+}
+
+TRIGGER_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_DEVICE_ID): str,
+            vol.Required(CONF_DOMAIN): DOMAIN,
+            vol.Required(CONF_PLATFORM): "device",
+            vol.Required(CONF_TYPE): str,
+            vol.Required(CONF_SUBTYPE): str,
+        }
+    )
+)
+
+
+def _get_deconz_event_from_device_id(hass, device_id):
+    """Resolve deconz event from device id."""
+    deconz_config_entries = configured_gateways(hass)
+    for config_entry in deconz_config_entries.values():
+
+        gateway = get_gateway_from_config_entry(hass, config_entry)
+        for deconz_event in gateway.events:
+
+            if device_id == deconz_event.device_id:
+                return deconz_event
+
+    return None
+
+
+async def async_attach_trigger(hass, config, action, automation_info):
+    """Listen for state changes based on configuration."""
+    config = TRIGGER_SCHEMA(config)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(config[CONF_DEVICE_ID])
+
+    trigger = (config[CONF_TYPE], config[CONF_SUBTYPE])
+
+    if device.model not in REMOTES and trigger not in REMOTES[device.model]:
+        return
```

Comment:
We should raise an exception when this happens. Maybe a new `InvalidDeviceAutomationConfig` exception ?